### PR TITLE
feat: Add authentication middleware

### PR DESCRIPTION
**Pull request by NikolayVerstak**
###  JWT Authentication Middleware

Implemented `verifyToken` middleware for JWT authentication (`middlewares/auth.middleware.js`). This middleware gets a token and an array of required roles as parameters for an endpoint to validate the token's authenticity and ensure that the user has access to those roles.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,5 @@
 PORT=
 
-MONGO_CONNECTION_STRING=
\ No newline at end of file
+MONGO_CONNECTION_STRING=
+
+SECRET_KEY=Autumn-fairytale
```
Comment (Am4rel) [inline .env.example:5]: do not store actual keys here, say `<string>` instead to clarify the format and communicate the actual key value to the team in chat

File: middlewares/auth.middleware.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * });
+ *
+ * @returns {string} - Required roleId ('userId', 'chefId', etc.) if the token is valid.
+ */
+
+const verifyToken = ({ role }) => {
+  const getTokenFromHeaders = (req) => req.headers.authorization?.split(' ')[1];
+
+  const getRoleId = (userRoles) =>
+    userRoles.find((userRole) => userRole.name === role)?.id;
```
Comment (Am4rel) [inline middlewares/auth.middleware.js:31]: `name === id`?
there should be an error either in params naming or in logic, doesn't `id` identify an `_id` of other collection?

File: middlewares/auth.middleware.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  return async (req, res, next) => {
+    try {
+      const token = getTokenFromHeaders(req);
+      if (!token) throw new UnAuthorizedError();
+
+      const { id } = jwt.verify(token, SECRET_KEY);
+
+      const user = await User.findById(id);
```
Comment (Am4rel) [inline middlewares/auth.middleware.js:40]: better to use `.exec()` on mongoose queries
https://mongoosejs.com/docs/promises.html#queries-are-not-promises

File: middlewares/auth.middleware.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import jwt from 'jsonwebtoken';
+import {
+  UnAuthorizedError,
+  NotFoundError,
+  ForbiddenError,
+} from '../helpers/errors.js';
+import { roles } from '../constants/index.js';
```
Comment (Am4rel) [inline middlewares/auth.middleware.js:7]: do we have it already?

File: middlewares/auth.middleware.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      const roleId = getRoleId(user.roles);
+      if (!roleId)
+        throw new ForbiddenError(
+          `User with id ${id} doesn't have a '${role}' account`
+        );
+
+      req[`${role}Id`] = roleId; // Generates "req.chefId", "req.courierId", "req.adminId"
+      next();
+    } catch (error) {
+      console.error(`Token verification error: ${error.message}`);
```
Comment (Am4rel) [inline middlewares/auth.middleware.js:57]: don't we have `winston` set up yet?

File: middlewares/auth.middleware.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      return res
+        .status(error.code || 401)
+        .send({ message: error.message || 'Invalid token' });
```
Comment (Am4rel) [inline middlewares/auth.middleware.js:60]: I'd suggest  to change default error message from `'Invalid token'` to something more neutral, we can potentially have other issues than token to cause an error here

**Review by Am4rel — CHANGES_REQUESTED** [review]
review the comments left

File: middlewares/auth.middleware.js (hunk 4 above)
Comment (NikolayVerstak) [reply middlewares/auth.middleware.js:7]: My mistake, agree) I'll fix it. Thank you

File: middlewares/auth.middleware.js (hunk 3 above)
Comment (NikolayVerstak) [reply middlewares/auth.middleware.js:40]: Thanx for info! I'll fix it

File: middlewares/auth.middleware.js (hunk 5 above)
Comment (NikolayVerstak) [reply middlewares/auth.middleware.js:57]: Agree. This line's useless

File: middlewares/auth.middleware.js (hunk 6 above)
Comment (NikolayVerstak) [reply middlewares/auth.middleware.js:60]: I got it, maybe  'Authentication failed' will be better

File: .env.example (hunk 1 above)
Comment (NikolayVerstak) [reply .env.example:5]: Sure, I got it

**Review by kyzym — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (NikolayVerstak) [reply .env.example]: Fixed 
https://github.com/Autumn-fairytale/AFT-back/pull/34/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR6

File: middlewares/auth.middleware.js (hunk 3 above)
Comment (NikolayVerstak) [reply middlewares/auth.middleware.js]: Fixed 
https://github.com/Autumn-fairytale/AFT-back/pull/34/files#diff-8018e43a672f37668be0b873bd5cb66267eba0256327671fa14313fe8ff17bb9R50

**Review by Am4rel — APPROVED** [review]

**Pull request merged**